zephyr/include/sof/trace/preproc.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2024 Intel Corporation.
+ */
+
+#ifndef __SOF_TRACE_PREPROC_H__
+#define __SOF_TRACE_PREPROC_H__
+
+#ifdef CONFIG_TRACE
+#error "SOF dma-trace (CONFIG_TRACE) not supported in Zephyr builds. \
+	Please use CONFIG_ZEPHYR_LOG instead."
+#endif
+
+#ifndef CONFIG_ZEPHYR_LOG
+
+/*
+ * Unlike the XTOS version of this macro, this does not silence all
+ * compiler warnings about unused variables.
+ */
+#define SOF_TRACE_UNUSED(arg1, ...) ((void)arg1)
```

Comment:
doesn't silence all - is it because of the lack of recursion? Leaving the rest of the arguments without `void`-casting? Should that be done instead?

Reply:
@lyakh Ack, in "OPTION A" (9598) there's "full silencing implementation". preproc-*.h does just this and question is whether to move that out of XTOS code or remote, and Liam was not keen to have this in SOF main library. I'd say, if you want full silencing, then just pull in preproc-*h, it's does just the thing, has the recursion helpers to iterate through the varargs in preprocessor.

Reply:
the first argument to `SOF_TRACE_UNUSED()` is the trace context, right? That's why we have to "silence" it here, because it isn't obvious where we actually call `comp_dbg()` and others. And other arguments we'll cast to `void` at their locations?

Reply:
@lyakh  This is really a very niche scenario of building with Zephyr but disable Zephyr logs (and fully disable, not just limiting logs to error level or something more usual). Now in Zephyr, logs are disabled like

```
#define LOG_DBG(...) (void) 0 
```

In "old SOF", there has been more elaborate macro magic that disables unused variable warnings even if a local variable is only used in "comp_dbg(comp, "print local variable foo %d", foo)" statements. Now we could handle this in multiple ways:
1) bring in the metaprocessor logic to support above kind of use (my OPTION A PR)
2) use __maybe_unused and be aligned how Zephyr debugging stameemnts work (this PR)
3) just live with the unsused variable warnings (currently treated as errors in our build and will fail CI)

I guess there are more variants of (1), but I'd say it's not worth the effort to make this pretty (especially as Zephyr does not do it).


raio. You are building with Zephyr but with Zephyr logging subsystem disabled (and fully disabled, not just limiting to plain error logs like we do on some targets). Zephyr headers take care of the